### ISSUE TITLE
Running on Ubuntu 20.04 ./command/target/release/command: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by ./command/target/release/command)

### ISSUE THREAD

**Issue by vlad-burca** [lichess-org/lila-docker#114]
I get this error when I try to start the service:
sudo ./lila-docker start

✓ Running on Docker Engine
    Updating crates.io index
 Downloading crates ...
  Downloaded either v1.13.0
  Downloaded equivalent v1.0.1
  Downloaded number_prefix v0.4.0
  Downloaded serde_spanned v0.6.6
  Downloaded lazy_static v1.5.0
  Downloaded zeroize_derive v1.4.2
  Downloaded smawk v0.3.2
  Downloaded zeroize v1.8.1
  Downloaded unicode-linebreak v0.1.5
  Downloaded thiserror-impl v1.0.63
  Downloaded serde_derive v1.0.204
  Downloaded ryu v1.0.18
  Downloaded unicode-ident v1.0.12
  Downloaded textwrap v0.16.1
  Downloaded serde v1.0.204
  Downloaded toml_edit v0.22.16
  Downloaded serde_json v1.0.120
  Downloaded winnow v0.6.14
  Downloaded syn v2.0.71
  Downloaded unicode-width v0.1.13
  Downloaded libc v0.2.155
  Downloaded syn v1.0.109
  Downloaded portable-atomic v1.7.0
  Downloaded cliclack v0.2.5
  Downloaded memchr v2.7.4
  Downloaded toml v0.8.15
  Downloaded indicatif v0.17.8
  Downloaded indexmap v2.2.6
  Downloaded hashbrown v0.14.5
  Downloaded proc-macro2 v1.0.86
  Downloaded quote v1.0.36
  Downloaded log v0.4.22
  Downloaded local-ip-address v0.6.1
  Downloaded neli v0.6.4
  Downloaded console v0.15.8
  Downloaded toml_datetime v0.6.6
  Downloaded itoa v1.0.11
  Downloaded thiserror v1.0.63
  Downloaded once_cell v1.19.0
  Downloaded neli-proc-macros v0.1.3
  Downloaded byteorder v1.5.0
    Finished `release` profile [optimized] target(s) in 0.85s
./command/target/release/command: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by ./command/target/release/command)
./command/target/release/command: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.32' not found (required by ./command/target/release/command)
./command/target/release/command: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by ./command/target/release/command)

**Comment by Carbrex** [comment]
Check your glibc version using ```ldd --version```.
And also try deleting target dir using ```rm -rf command/target``` and running ```./lila-docker start``` again.

**Comment by vlad-burca** [comment]
`ldd --version`
```
ldd (Ubuntu GLIBC 2.31-0ubuntu9.16) 2.31
Copyright (C) 2020 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
Written by Roland McGrath and Ulrich Drepper.
```

**Comment by Carbrex** [comment]
Try deleting target dir using rm -rf command/target and running ./lila-docker start again.

**Comment by vlad-burca** [comment]
I tried doing  rm -rf command/target and running ./lila-docker start but I get the same result.

**Comment by vlad-burca** [comment]
`sudo apt update`
Hit:1 http://us-east-1.ec2.archive.ubuntu.com/ubuntu focal InRelease
Get:2 http://us-east-1.ec2.archive.ubuntu.com/ubuntu focal-updates InRelease [128 kB]                                                                                                                   
Hit:3 http://us-east-1.ec2.archive.ubuntu.com/ubuntu focal-backports InRelease                                                                                                                          
Get:4 http://security.ubuntu.com/ubuntu focal-security InRelease [128 kB]                                                                                                                               
Hit:5 https://download.docker.com/linux/ubuntu focal InRelease                                                                                                                                          
Ign:6 https://repo.mongodb.org/apt/ubuntu focal/mongodb-org/5.0 InRelease                                             
Hit:7 https://repo.mongodb.org/apt/ubuntu focal/mongodb-org/5.0 Release                         
Get:8 https://deb.nodesource.com/node_17.x focal InRelease [4583 B]       
Hit:9 https://scala.jfrog.io/artifactory/debian all InRelease             
Get:11 http://us-east-1.ec2.archive.ubuntu.com/ubuntu focal-updates/main amd64 Packages [3536 kB]
Ign:10 https://scala.jfrog.io/artifactory/debian  InRelease
Get:12 http://us-east-1.ec2.archive.ubuntu.com/ubuntu focal-updates/main Translation-en [547 kB]
Hit:13 https://scala.jfrog.io/artifactory/debian  Release                             
Get:14 http://security.ubuntu.com/ubuntu focal-security/main amd64 Packages [3164 kB]
Get:15 http://security.ubuntu.com/ubuntu focal-security/main Translation-en [468 kB]
Fetched 7974 kB in 3s (2692 kB/s)                               
Reading package lists... Done
Building dependency tree       
Reading state information... Done
All packages are up to date.

`sudo apt upgrade`
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Calculating upgrade... Done
Get more security updates through Ubuntu Pro with 'esm-apps' enabled:
  redis-tools redis-server
Learn more about Ubuntu Pro on AWS at https://ubuntu.com/aws/pro
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.

I think this is the latest version for ubuntu 20.04

**Comment by Carbrex** [comment]
If you uninstall your local rust it would run the rust code in a docker container, maybe then it would work.

**Comment by vlad-burca** [comment]
I removed rust 
`rustc --version`
-bash: /usr/bin/rustc: No such file or directory
and `rm -rf command/target` but still the same result.

**Comment by fitztrev** [comment]
do you get any output from these?

```bash
rustup --version
cargo --version
```

if it's still trying to use rust from your host, try removing this part of the conditional

https://github.com/lichess-org/lila-docker/blob/5ebd06bdeb4f4eaabec8c97db8c0a604e48c5f07/lila-docker#L173-L180

**Comment by paxvinci** [comment]
I receive the same error. I made a fresh machine with Ubuntu 20.04. I made update & upgrade, installed docker, cloned the repo and I followed the instructions. Tried to clean and reinstall. Same problem. The ldd is the same as above reported. I don't have rust or cargo installed on the VM.

**Comment by paxvinci** [comment]
> > made a fresh machine
> 
> > installed on the VM
> 
> what is your environment? I'd recommend running Docker natively if you can.

Ubuntu 20.04, upgraded. Docker version 27.3.1, build ce12230. I installed the VM less than 1 hour ago so new environment. The VM is hosted in an local Hypervisor (proxmox).

**Comment by fitztrev** [comment]
@vlad-burca are you also trying to run in a virtualized environment?

**Comment by vlad-burca** [comment]
No, I was running it on and aws ec2 instance. But I closed the task, suggesting an update to the instance. But thank you for your support 😉
